### DISCUSSION TITLE
Running Deep-Live-Cam on Raspberry Pi 5

### DISCUSSION THREAD

**Discussion by omer-re** [hacksider/Deep-Live-Cam#453] [Show and tell]
After succeeding running this cool project it on a Macbook, I have tried making it work on my RPI5+AI kit.
It took me the weekend, but I was eventually able to make it run, even if very laggy:
![image](https://github.com/user-attachments/assets/b7a6b581-cef0-4fb6-9603-e0aab08b7b89)

This wonderful project is able to run on "simple" machines, not just heavily reinforced with GPU. 
As far as I understand, it unfortunately not harnessing any of the Hailo 8L abilities.
Are there members here that have experimented with such libraries and applications over the Pi5+8L kit?

### What I did I order to run it on the Raspberry pi?
This is the state of my env that allowed me to run it. I did the minimal changes required to make it work, and I am sure it can be done better or more efficient.

##### 1. Set up
```
sudo apt-get install -y build-essential checkinstall libreadline-gplv2-dev libncursesw5-dev libssl-dev libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev zlib1g-dev openssl libffi-dev liblzma-dev libgmp-dev libmpdec-dev libmpfr-dev libmpc-dev libdb-dev uuid-dev  libffi-dev
sudo apt-get install python3-tk
sudo apt-get install tk-dev tcl-dev
```

##### 2. Install python 3.10 

```
wget https://www.python.org/ftp/python/3.10.0/Python-3.10.0.tgz
tar -xvf Python-3.10.0.tgz
cd Python-3.10.0

./configure --enable-optimizations --with-tcltk
make -j $(nproc)
sudo make install
```

##### 3. Create venv, install adjusted requirements
```
python3 -m venv --system-site-packages dlc310
source dlc3103/bin/activate
/home/hailopi/Deep-Live-Cam/dlc3103/bin/python3 -m pip install --upgrade pip
pip install -r rpi5_requirements.txt

python run.py
```

with `rpi5_requirements.txt` as follows:

```

absl-py==2.1.0
addict==2.4.0
astunparse==1.6.3
basicsr @ git+https://github.com/xinntao/BasicSR.git@8d56e3a045f9fb3e1d8872f92ee4a4f07f886b0a
certifi==2024.7.4
charset-normalizer==3.3.2
coloredlogs==15.0.1
contourpy==1.2.1
cycler==0.12.1
facexlib==0.3.0
filelock==3.15.4
filterpy==1.4.5
flatbuffers==20181003210633
fonttools==4.53.1
fsspec==2024.6.1
future==1.0.0
gast==0.6.0
gfpgan @ git+https://github.com/TencentARC/GFPGAN.git@7552a7791caad982045a7bbe5634bbf1cd5c8679
google-pasta==0.2.0
grpcio==1.66.0
h5py==3.11.0
humanfriendly==10.0
idna==3.8
imageio==2.35.1
insightface==0.7.3
importlib_metadata==8.4.0
Jinja2==3.1.4
keras==3.5.0
kiwisolver==1.4.5
lazy_loader==0.4
libclang==18.1.1
llvmlite==0.43.0
lmdb==1.5.1
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.9.2
mdurl==0.1.2
ml-dtypes==0.4.0
mpmath==1.3.0
namex==0.0.8
networkx==3.3
numba==0.60.0
numpy==1.26.4
onnxruntime==1.16.3
opencv-python==4.10.0.84
opt-einsum==3.3.0
optree==0.12.1
packaging==24.1
pillow==10.4.0
platformdirs==4.2.2
protobuf==4.25.4
Pygments==2.18.0
pyparsing==3.1.2
python-dateutil==2.9.0.post0
PyYAML==6.0.2
requests==2.32.3
rich==13.7.1
scikit-image==0.24.0
scipy==1.14.1
six==1.16.0
sympy==1.13.2
tb-nightly==2.18.0a20240824
tensorboard==2.17.1
tensorboard-data-server==0.7.2
tensorflow==2.17.0
tensorflow-io-gcs-filesystem==0.37.1
termcolor==2.4.0
tifffile==2024.8.10
tomli==2.0.1
torch==2.4.0
torchvision==0.19.0
tqdm==4.66.5
typing_extensions==4.12.2
urllib3==2.2.2
Werkzeug==3.0.4
wrapt==1.16.0
yapf==0.40.2
zipp==3.20.0
customtkinter==5.2.2
```

Even though it runs slow, I think it is amazing that such simple hardware is able to run it, and it is equally impressive to make the project that simple to deploy. I am sure it can be better adapted to the Pi (especially if using Hailo's Pi 5 AI kit).
If anyone has some experience doing that- please comment below 🤓

https://github.com/user-attachments/assets/c939de1d-c91d-4ada-a5d9-d41d2a4db509

**Comment by hacksider** [comment]
That's amazing ☺️

**Reply by djh221** [reply]
Can the blogger help me see that the photos that come out of my code when it runs are blurry

**Comment by djh221** [comment]
博主能不能帮我看一下，我的代码运行出来的图片很模糊

**Comment by harshb20** [comment]
As far as my understanding goes, if at all there would be next to none benefit with the current Hailo NPU, primarily not because of any potential abnormal amounts of bugs or the TOPS of hardware, but due to the memory bandwidth, the it uses just 1x pcie slot for communication and even in it's best form (pcie 3.0) chances are that just the data transfer from cpu to npu would be a big bottleneck, negating it as a feasible solution(again, like I haven't explored much specifics of to be cent percent sure but this is a big consideration.......)

**Reply by UltraSeeder** [reply]
Hey, how did you get it to run on MacBook. I'm using an M1 and the errors just keep popping up! Would love to know, or do like you did here.
